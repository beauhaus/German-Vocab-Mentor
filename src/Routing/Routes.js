import React from 'react';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';
import styled from 'styled-components';

import MainNav from './MainNav';
import Template from '../components/Template';
import Template2 from '../components/Template2';
import Template3 from '../components/Template3';
import Template4 from '../components/Template4';
import FourOhFour from './FourOhFour';

// const db = getLS();

// ReactDOM.render(<Landing db={db}/>, document.getElementById('app'));
const RouterLiner = styled.div`
    width: 100wv;
    height: 100vh;
    /* height: auto; */
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: repeat(20, 1fr);
    /* border: 1px solid orange; */
`;


const Routes = (props) => (
    <BrowserRouter>
        <RouterLiner className="router-liner">
            <MainNav />
            <Switch>
                <Route path="/" exact={true} />
                <Route path="/template" component={Template} />
                <Route path="/template2" component={Template2} />
                <Route path="/template3" component={Template3} />
                <Route path="/template4" component={Template4} />
                <Route component={FourOhFour} />
            </Switch>
        </RouterLiner>
    </BrowserRouter>
)

module.exports = Routes;