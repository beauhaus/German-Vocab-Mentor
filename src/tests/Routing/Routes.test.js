//react-test-renderer
import React from 'react';
import {shallow} from 'enzyme';
import Routes from '../../Routing/Routes'
// import toJSON from 'enzyme-to-json';


/****** ENZYME METHODS FOR SHALLOW RENDER: ******

https://airbnb.io/enzyme/docs/api/shallow.html

************************************************/

test('should render routes correctly',() => {
    const wrapper = shallow(<Routes/>)
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.find('h1').text()).toBe("Hello")
    // const renderer = new ReactShallowRenderer;
    // renderer.render(<Routes/>)
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
    // console.log("R: \n", renderer.getRenderOutput())
})