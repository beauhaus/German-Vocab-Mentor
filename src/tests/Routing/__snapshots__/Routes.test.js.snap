// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render routes correctly 1`] = `
<BrowserRouter>
  <.sc-EHOje
    className="router-liner"
  >
    <MainNav />
    <Switch>
      <Route
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/template"
      />
      <Route
        component={[Function]}
        path="/edit:id"
      />
      <Route
        component={[Function]}
        path="/addVocabPage"
      />
      <Route
        component={[Function]}
        path="/vocabItemDashboard"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </.sc-EHOje>
</BrowserRouter>
`;
